src/metadata/parse.rs

```diff
@@ -11,9 +13,137 @@ pub fn parse_date(val: &str) -> ParseResult {
+        // link reg
+        static ref LINK_REG: Regex = Regex::new(r"^\w+:").unwrap();
+        // email reg
+        static ref EMAIL_REG: Regex = Regex::new(r".+@.+\..+").unwrap();
+    }
+
+    fn is_linkish(piece: &str) -> bool {
+        LINK_REG.is_match(piece)
+    }
+
+    fn is_emailish(piece: &str) -> bool {
+        EMAIL_REG.is_match(piece)
+    }
+
+    // TODO: unescape pieces
+    let mut pieces = val
+        .split(",")
+        .map(|piece| piece.trim())
+        .collect::<Vec<&str>>();
+
+    if pieces.is_empty() {
+        return Err("invalid editor");
+    }
+
+    let mut editor = Editor::new(pieces[0].to_owned());
+
+    // extract w3c id
+    pieces = pieces[1..]
+        .iter()
+        .cloned()
+        .filter(|piece| {
+            if W3C_ID_REG.is_match(piece) && editor.w3c_id.is_none() {
+                editor.w3c_id = Some((&piece[6..]).to_owned());
+                false
+            } else {
+                true
+            }
+        })
+        .collect::<Vec<&str>>();
+
+    // handle ambiguous pieces
+    if pieces.len() == 3
+        && ((is_emailish(pieces[1]) && is_linkish(pieces[2]))
+            || (is_linkish(pieces[1]) && is_emailish(pieces[2])))
+    {
+        // [org, email, link] or [org, link, email]
+        editor.org = Some(pieces[0].to_owned());
+        if is_emailish(pieces[1]) {
+            editor.email = Some(pieces[1].to_owned());
+            editor.link = Some(pieces[2].to_owned());
+        } else {
+            editor.link = Some(pieces[1].to_owned());
+            editor.email = Some(pieces[2].to_owned());
+        }
+    } else if pieces.len() == 2
+        && ((is_emailish(pieces[0]) && is_linkish(pieces[1]))
+            || (is_linkish(pieces[0]) && is_emailish(pieces[1])))
+    {
+        // [email, link] or [link, email]
+        if is_emailish(pieces[0]) {
+            editor.email = Some(pieces[0].to_owned());
+            editor.link = Some(pieces[1].to_owned());
+        } else {
+            editor.link = Some(pieces[0].to_owned());
+            editor.email = Some(pieces[1].to_owned());
+        }
+    } else if pieces.len() == 2 && (is_emailish(pieces[1]) || is_linkish(pieces[1])) {
+        // [org, email] or [org, link]
+        editor.org = Some(pieces[0].to_owned());
+        if is_emailish(pieces[1]) {
+            editor.email = Some(pieces[1].to_owned());
+        } else {
+            editor.link = Some(pieces[1].to_owned());
+        }
+    } else if pieces.len() == 1 {
+        // [org], [email], or [link]
+        if is_emailish(pieces[0]) {
+            editor.email = Some(pieces[0].to_owned());
+        } else if is_linkish(pieces[0]) {
+            editor.link = Some(pieces[0].to_owned());
+        } else {
+            editor.org = Some(pieces[0].to_owned());
+        }
+    } else if pieces.len() != 0 {
+        return Err("wrong format");
+    }
+
+    // check if the org ends with an email or a link
+    if let Some(org) = editor.org.clone() {
```

Comment:
I think you could avoid the clone here with some gymnastics, but probably not worth it.